Userland/Libraries/LibWeb/HTML/CanvasRenderingContext2D.cpp

```diff
@@ -76,6 +77,7 @@ void CanvasRenderingContext2D::fill_rect(float x, float y, float width, float he
         auto rect = drawing_state.transform.map(Gfx::FloatRect(x, y, width, height));
         auto color_fill = drawing_state.fill_style.as_color();
         if (color_fill.has_value()) {
+            color_fill.value().set_alpha(drawing_state.global_alpha * 255);
```

Comment:
Playing around with some examples in other browsers, I don't think this is correct.
It looks like it should actually be `color.set_alpha(color.alpha() *  drawing_state.global_alpha)`. I added a convenience function `color.with_opacity()` to do that in #18927, so maybe this should use that (if that's merged).